envs/monkey_zoo/blackbox/analyzers/communication_analyzer.py

```diff
@@ -13,15 +14,22 @@ def __init__(self, island_client: MonkeyIslandClient, machine_ips: Iterable[str]
 
     def analyze_test_results(self):
         self.log.clear()
-        all_monkeys_communicated = True
+        all_agents_communicated = True
+        agent_ips = self._get_agent_ips()
+
         for machine_ip in self.machine_ips:
-            if not self.did_monkey_communicate_back(machine_ip):
-                self.log.add_entry("Monkey from {} didn't communicate back".format(machine_ip))
-                all_monkeys_communicated = False
+            if self._agent_communicated_back(machine_ip, agent_ips):
+                self.log.add_entry("Agent from {} communicated back".format(machine_ip))
             else:
-                self.log.add_entry("Monkey from {} communicated back".format(machine_ip))
-        return all_monkeys_communicated
+                self.log.add_entry("Agent from {} didn't communicate back".format(machine_ip))
+                all_agents_communicated = False
+
+        return all_agents_communicated
```

Comment:
Approved.

If we're nit picking, you can short-circuit the loop.

```suggestion
                self.log.add_entry("Agent from {} didn't communicate back".format(machine_ip))
                return False

        return True
```

Then remove line 17.